### ISSUE TITLE
Region average - grid resolution (various related problems)

### ISSUE THREAD

**Issue by ET-NCMP** [ET-NCMP/NCMP#29]
As written, the code uses a 2 degree grid. Is this “too coarse”? Testing for Australia would suggest so, as the method applied to define “land” (see the next point) excluded many coastal 2 degree grid boxes where ACORN-SAT land stations are located. What is “land” does affect other global temperature analyses, for example the GISS land-only temperature analysis (also on a 2 degree grid, but in this case there is possibly also the definition of what GISS considers to be a “land” temperature station).

The kriging example in the User Guide used a 0.1 degree grid, and noted that using too coarse a grid will lead to errors in the estimation of the area average because too great a weight will be given to coastal areas. That would depend on how “land” is defined – if coastal areas are too easily excluded, wouldn’t the error would be due to the coast having too little weight?

The more fundamental issue is how the land grid boxes are defined. This involves checking which (2 degree) grid box centre lies inside polygons of a coastline database, usually subsetted to include the correct country. Testing of this procedure suggested that a 2 degree grid, constrained to integer values, will fail to find any land point for about one-third of the 240+ valid country codes. As noted above, it does not guarantee that a station will lie inside a valid land grid box. Since the “country” is a subset of “land” (within the coastline database), this issue also affects the code’s “regional average” option. One solution is using a higher resolution grid, for example 1 degree as used by the Berkeley Earth spatial analysis (which also uses a form of kriging). But this may not solve the issue of defining “land” points, and will result in a four-fold increase in computational time. Another solution is to define any grid box containing a station as “land”, but this may also not be “correct”, e.g. possibly GISS land-only temperature. It seems probable that greater user flexibility is necessary to appropriately define the “land” points for their country/region.

**Comment by ET-NCMP** [comment]
I've added a user-defined grid size to the options that need to be entered to run P4_Region_Average. Users should probably stick to one of a small number of resolutions, to keep things simple.

A low resolution can be used for testing, to make sure everything runs OK, but a higher resolution should be used for the final calculation because of the coastline/border problem that Simon noted.

The amount of time it takes to calculate the grids is fairly linear with the number of grid cells and so halving the grid-square size will quadruple the time it takes to run, which could be prohibitive in extreme cases (large domain, complex coastlines). I ran Canada, which is largish and complex at 0.25 degree resolution which took maybe half an hour on my desktop machine. I run the UK at 0.1 degree resolution and that's fast.

0.1 degrees corresponds to about 11km or less, so I don't think it will be necessary to go to higher resolution, but I'm not 100% sure.
